### ISSUE TITLE
[docs] clarify the usage of `versionadded` & co

### ISSUE THREAD

**Issue by picnixz** [sphinx-doc/sphinx#12412]
### Describe the bug

> It's interesting to see that the `::` construction fails when inside the `versionadded/versionchanged` directive. It might actually be an issue in Sphinx (because otherwise, the `::` should be smart). 

_Originally posted in https://github.com/python/cpython/issues/120091#issuecomment-2150319649_

I'm not really sure how to make it work but if it's too much work, let's just indicate the behaviour in the documentation of `.. versionadded` & co directives.

### How to Reproduce

<details>

```rst
Welcome to documentation!
=========================


It is now::

>>> ok

It is now2::

   >>> ok

.. versionadded:: 3.14 It is
   now3::

   >>> ok

.. versionadded:: 3.14 It is
   now4::

      >>> ok


.. versionadded:: 3.14
   It is now5::

   >>> ok


.. versionadded:: 3.14
   It is now6::

      >>> ok
```

produces 

![image](https://github.com/sphinx-doc/sphinx/assets/10796600/9f410e46-ebad-481d-ad2a-8ae996ef4b9d)

</details>

**Comment by picnixz** [comment]
Ah yes! it's an argument! I totally forgot about that for those directives (I think it's the only directive that does this kind of thing). Should we then perhaps say that `::` is no more smart in this case?

**Comment by picnixz** [comment]
Wait.. now I'm confused, adding a blank line before seems to work (and does what I would have expected) but it's not correct according to the docs.

**Comment by picnixz** [comment]
Yes, but with the body... it works for me locally. And the HTML is consistent so I think we already did the change but we did not mention it on the docs? We haven't touched that code since 2018 roughly.

**Comment by chrisjsewell** [comment]
ok yeh it does have a body, but that it parsed separately to the (optional) first paragraph: https://github.com/sphinx-doc/sphinx/blob/b90482fc8147dc7d171c07897e11472fb1ed2462/sphinx/domains/changeset.py#L67

But yeh, if you want to use a body, I feel we should definitely be telling people to use it as the body, rather than the same line as the version

**Comment by picnixz** [comment]
So... I'm not sure what should be supported but we seem to support the 2 args + body. The second argument is directly put after the version changed, whether you put that second argument on the same line as the version *or* on the next line (without a blankline), e.g.:

```rst
.. versionchanged:: 3.14 SHORT_DESC

.. versionchanged:: 3.14 
   SHORT_DESC
```

If you want a body (i.e., something that would be visually on the next line), you can put one **but** it will be aligned with the "Changed in version [...]" and not indented, e.g.:

```rst
.. versionadded:: 3.14
   ok

   See::

      1 + 1

   and then
```

gives you

![image](https://github.com/sphinx-doc/sphinx/assets/10796600/2fa7c504-26bc-4c24-b40b-2a9029efd416)

On the other hand,


```rst
.. versionadded:: 3.14
   okokok

      See::

         1 + 1

      and then
```

gives you

![image](https://github.com/sphinx-doc/sphinx/assets/10796600/4dc53ba9-80d1-4fb6-957b-300d498a28fc)

versus

**Comment by picnixz** [comment]
I don't know what to recommend now. I think it's just better to say "put whatever you want as a description as a body, whether it is a short description or not" but still, we should mention the alignment of the paragraph.

**Comment by chrisjsewell** [comment]
> put whatever you want as a description as a body, whether it is a short description or not

yep 👍 
the allowing of the body in the argument feels an over-complication, for the minimal benefit of making things a little less terse; I never actually used it like this, as I didn't even know it was allowed
